src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -305,6 +305,35 @@ public override async Task<bool> CheckHealthAsync(CancellationToken cancellation
             }
         }
 
+        public override async Task<bool> CheckOutboundHealthAsync(CancellationToken cancellationToken = default)
+        {
+            var path = "/v1.0/healthz/outbound";
+            var request = new HttpRequestMessage(HttpMethod.Get, new Uri(this.httpEndpoint, path));
+            try
+            {
+                var response = await this.httpClient.SendAsync(request, cancellationToken);
+                return response.IsSuccessStatusCode;
+            }
+            catch (HttpRequestException)
+            {
+                return false;
+            }
+        }
+
+        public override async Task WaitForSidecarAsync(CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            while (true)
+            { 
+                var response = await CheckOutboundHealthAsync(cancellationToken);
+                if (response)
+                {
+                    break;
+                }
+                await Task.Delay(TimeSpan.FromMilliseconds(500));
```

Comment:
Change this to https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?view=net-6.0#system-threading-tasks-task-delay(system-int32-system-threading-cancellationtoken) and then this will allow you to cancel immediately out of the `Task.Delay`. 

This way you don't have to check the cancellation token manually in this method because all of *the work* also supports cancellation.

Reply:
Nifty.